File: Fragment.Fragment/MyFlexibleFragment/java/app/src/main/java/com/dicoding/picodiploma/myflexiblefragment/MainActivity.java
package com.dicoding.picodiploma.myflexiblefragment;

import android.os.Bundle;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;
import androidx.appcompat.app.AppCompatActivity;
import android.util.Log;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        FragmentManager mFragmentManager = getSupportFragmentManager();
        FragmentTransaction mFragmentTransaction = mFragmentManager.beginTransaction();

        HomeFragment mHomeFragment = new HomeFragment();

        Fragment fragment = mFragmentManager.findFragmentByTag(HomeFragment.class.getSimpleName());

        /*
        Kode di bawah ini digunakan untuk memvalidasi apakah suatu fragment adalah instance dari suatu kelas
         */
        if (!(fragment instanceof HomeFragment)) {
            mFragmentTransaction.add(R.id.frame_container, mHomeFragment, HomeFragment.class.getSimpleName());

            Log.d("MyFlexibleFragment", "Fragment Name :" + HomeFragment.class.getSimpleName());

            mFragmentTransaction.commit();
        }
    }
}

File: Fragment.Fragment/MyFlexibleFragment/kotlin/app/src/main/java/com/dicoding/picodiploma/myflexiblefragment/MainActivity.kt
package com.dicoding.picodiploma.myflexiblefragment

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val mFragmentManager = supportFragmentManager
        val mFragmentTransaction = mFragmentManager.beginTransaction()

        val mHomeFragment = HomeFragment()

        val fragment = mFragmentManager.findFragmentByTag(HomeFragment::class.java.simpleName)

        /*
        Kode di bawah ini digunakan untuk memvalidasi apakah suatu fragment adalah instance dari suatu kelas
         */
        if (fragment !is HomeFragment) {
            mFragmentTransaction.add(R.id.frame_container, mHomeFragment, HomeFragment::class.java.simpleName)

            Log.d("MyFlexibleFragment", "Fragment Name :" + HomeFragment::class.java.simpleName)

            mFragmentTransaction.commit()
        }
    }
}
